### PR TITLE
Add EM Mito-Nuc Generalist

### PR DESCRIPTION
**Pull request by anwai98**

### CODE REVIEW

File: finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+from torch_em.data import ConcatDataset, MinInstanceSampler, datasets
+
+from micro_sam.training import identity
+
+
+class ResizeRawTrafo:
+    def __init__(self, desired_shape, do_rescaling=True, padding="constant"):
+        self.desired_shape = desired_shape
+        self.padding = padding
+        self.do_rescaling = do_rescaling
```
Comment (constantinpape) [inline finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py:18]: This is not actually used. Is that intended? Should it be removed?

File: finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    platy_nuclei_train_dataset = platy_nuclei_dataset(platy_nuclei_train_rois, sample_ids=platy_nuclei_train_samples)
+    platy_nuclei_val_dataset = platy_nuclei_dataset(platy_nuclei_val_rois, sample_ids=platy_nuclei_val_samples)
+
+    train_datasets = [mitoem_train_dataset, platy_nuclei_train_dataset]
+    val_datasets = [mitoem_val_dataset, platy_nuclei_val_dataset]
+
+    if with_cem:
+        def cem_dataset(split):
+            return datasets.cem.get_mitolab_dataset(
+                path=os.path.join(input_path, "mitolab"), split="train", val_fraction=0.1
```
Comment (constantinpape) [inline finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py:136]: You have `split="train"` hard-coded here, but it should be `split=split`!!

File: finetuning/generalists/training/electron_microscopy/mito_nuc/train_mito_nuc_em_generalist.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    model.to(device)
+
+    # let's get the UNETR model for automatic instance segmentation pipeline
+    unetr = UNETR(
+        backbone="sam",
+        encoder=model.sam.image_encoder,
+        out_channels=3,
+        use_sam_stats=True,
+        final_activation="Sigmoid",
+        use_skip_connection=False
```
Comment (constantinpape) [inline finetuning/generalists/training/electron_microscopy/mito_nuc/train_mito_nuc_em_generalist.py:43]: It's probably best to set `resize_input=True` here. (This is True by default now, but it's good to be explicit here.)

**Review by constantinpape — CHANGES_REQUESTED** [review]
Found a typo in the CEM loader and two minor comments.

File: finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py (hunk 1 above)
Comment (anwai98) [reply finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py:18]: Yes, this isn't necessary. I will remove it. Thanks!

File: finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py (hunk 2 above)
Comment (anwai98) [reply finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py:136]: Fixed it, thanks!

File: finetuning/generalists/training/electron_microscopy/mito_nuc/train_mito_nuc_em_generalist.py (hunk 3 above)
Comment (anwai98) [reply finetuning/generalists/training/electron_microscopy/mito_nuc/train_mito_nuc_em_generalist.py:43]: Makes sense. Updated this as wlel. Thanks!

File: finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+from elf.io import open_file
+
+from torch_em import get_data_loader
+from torch_em.transform.label import PerObjectDistanceTransform
+from torch_em.data import ConcatDataset, MinInstanceSampler, datasets
+
+from micro_sam.training import identity
+
+
+class ResizeRawTrafo:
```
Comment (constantinpape) [inline finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py:14]: The trafos should be refactored, see comments in #292.

Comment (anwai98) [reply finetuning/generalists/training/electron_microscopy/mito_nuc/obtain_mito_nuc_em_datasets.py]: These have been taken care of now. Thanks!

**Review by constantinpape — APPROVED** [review]

**Pull request merged**